src/shared/install.h

```diff
@@ -74,7 +74,7 @@ enum UnitFileChangeType {
         UNIT_FILE_UNLINK,
         UNIT_FILE_IS_MASKED,
         _UNIT_FILE_CHANGE_TYPE_MAX,
-        _UNIT_FILE_CHANGE_TYPE_INVALID = -1
+        _UNIT_FILE_CHANGE_INVALID = INT_MIN
 };
```

Comment:
This is not entirely sufficient I fear... The call string_table_lookup() function actually returns "-1" on failure, which is cast to whatever enum is used for the DEFINE_STRING_TABLE_LOOKUP() macro invocaton. This means invoking unit_file_change_type_from_string("foo") will currently return -EPERM, and not _UNIT_FILE_CHANGE_INVALID, as it should.

Now, the thing is that we currently don't actually use unit_file_change_type_from_string(), the macro defines it for us though. I am bit concerned that we might eventually though and then this will explode in our faces...

I see three ways out: a) extend the string-table.h macro stuff so that it optoinally takes an additional argument that allows overriding which precise value is used for "invalid", so that we can pass INT_MIN rather than -1 to it in this case. Or b), stop using the string-table macros for this, or only for the translation in one direction. But of course, we might run into trouble with this later on then, should we ever need the other direction after all, and don't think about this then. Or c) ignore the issue, and just plaster everything with warning comments about this.

I am leaning towards b) I must say...